x-pack/plugins/case/server/routes/api/cases/helpers.ts

```diff
@@ -138,11 +142,13 @@ export const transformESConnectorToCaseConnector = (connector: ESCaseConnector):
   name: connector?.name ?? 'none',
   type: connector?.type ?? '.none',
   fields:
-    connector?.fields.reduce(
-      (fields, field) => ({
-        ...fields,
-        [field.key]: field.value,
-      }),
-      {}
-    ) ?? {},
+    connector.fields.length > 0
+      ? (connector.fields.reduce(
+          (fields, { key, value }) => ({
+            ...fields,
+            [key]: value,
+          }),
+          {}
+        ) as ConnectorFields)
```

Comment:
i know this is uncool but otherwise it thinks it possible to be `{}` even though L145 checks to make sure that there are indeed fields to reduce through...